cmd/otel-allocator/server/server.go

```diff
@@ -88,6 +88,7 @@ func NewServer(log logr.Logger, allocator allocation.Allocator, listenAddr strin
 	router.GET("/jobs", s.JobHandler)
 	router.GET("/jobs/:job_id/targets", s.TargetsHandler)
 	router.GET("/metrics", gin.WrapH(promhttp.Handler()))
+	router.GET("/healthz", s.LivenessProbeHandler)
```

Comment:
can you rename this to be `livez`?  that's what kube recommends :) https://kubernetes.io/docs/reference/using-api/health-checks/#api-endpoints-for-health